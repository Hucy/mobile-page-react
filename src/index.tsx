import React, { Fragment, SFC } from 'react';
import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  html,body{
    margin:0;
    padding:0;
  }
`;

export const MaxHeightContent = styled.div`
  height: 100%;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
`;

export const PageStyle = styled(MaxHeightContent)`
  position: absolute;
  width: 100%;
`;

interface IPage {
  className?: string;
}

const Page: SFC<IPage> = ({ className, children }) => (
  <Fragment>
    <GlobalStyle />
    <PageStyle className={className}>{children}</PageStyle>
  </Fragment>
);

export default Page;

export const FixedContent = styled.div`
  position: absolute;
  z-index: 1000;
`;

export const MainContent = styled.div`
  overflow: hidden;
  overflow-y: scroll;
  flex: 1;
  -webkit-overflow-scrolling: touch;
`;
