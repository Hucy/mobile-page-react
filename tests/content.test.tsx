import 'jest-styled-components';
import React from 'react';

import { mount } from 'enzyme';

import { FixedContent, MainContent, MaxHeightContent } from '../src';

test('should render correctly', () => {
  const FixedWrapper = mount(<FixedContent />);
  expect(FixedWrapper).toMatchSnapshot();
  const MainWrapper = mount(<MainContent />);
  expect(MainWrapper).toMatchSnapshot();

  const MaxHeightContentWrapper = mount(<MaxHeightContent />);
  expect(MaxHeightContentWrapper).toMatchSnapshot();
});
