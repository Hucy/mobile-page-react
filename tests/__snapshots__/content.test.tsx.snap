// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  position: absolute;
  z-index: 1000;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`should render correctly 2`] = `
.c0 {
  overflow: hidden;
  overflow-y: scroll;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-overflow-scrolling: touch;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  overflow: hidden;
  overflow-y: scroll;
  flex: 1;
  -webkit-overflow-scrolling: touch;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`should render correctly 3`] = `
.c0 {
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  height: 100%;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
